symphony/lib/toolkit/class.emailgateway.php

```diff
@@ -610,13 +631,15 @@ protected function getSectionAttachments()
                      . $this->contentInfoString($file['mime-type'], $file['file'], $file['filename'], $file['charset'])
                      . EmailHelper::base64ContentTransferEncode($file_content);
             } else {
-                if (!$tmp_file === false) {
-                    $filename = $original_filename;
-                } else {
-                    $filename = $file['file'];
+                if ($this->_ignore_attachment_errors === false) {
+                    if (!$tmp_file === false) {
```

Comment:
> But of course my setter function only allows boolean values—so should I use a "less strict" check here?

Yup. I do not mind embracing the dynamic aspect of PHP (or javascript for that matter). If it walks like a duck and sings like a duck, it's a duck.

> It might then simply be if ($this->_validate_attachment_errors) 

In fact!

> I like to learn, and you know better than me what is "good practice". So which version is better in this context?

Both have flaws and advantages. Being more readable (less verbose) vs being explicit. Being flexible vs. being performant. I would try to find what PSR tells about this.